test/src/input.tsx

```diff
@@ -0,0 +1,51 @@
+import { render, screen, fireEvent } from "@testing-library/react";
+import { Input } from "@moai/core";
+import { useState } from "react";
+
+const InputTesting = () => {
+	const [value, setValue] = useState("");
+	return (
+		<div>
+			<Input value={value} setValue={setValue} data-testid="input" />
+			<div data-testid="div">{value}</div>
+			<button data-testid="button" onClick={() => setValue("Goodbye")}>
+				Change
+			</button>
+		</div>
+	);
+};
+
+describe("Testing Input Component", () => {
+	test("div value should change after Input value change", async () => {
+		const spy = jest.spyOn(console, "error");
+		spy.mockImplementation(() => void 0);
+
+		render(<InputTesting />);
+
+		const inputElement = screen.getByTestId("input") as HTMLInputElement;
+		const divElement = screen.getByTestId("div");
```

Comment:
Most of the times we should try to have better way than "test-id" (even though it's common).

For the input, try to get it by a label: https://testing-library.com/docs/queries/bylabeltext/

For the div, we don't actually need to check for the text content of the div (it's too much an implementation detail). Instead, we should just check if the text is available on the screen, via something like https://testing-library.com/docs/queries/bytext after the fireEvent